### PR TITLE
fix: Resolve issues with deployment scripting

### LINKED ISSUE #102: Hook up deployment machinery
We have [a build script](https://github.com/bwinton/SnoozeTabs/blob/master/bin/build-addon.sh). It needs hooking up to [Test Pilot auto build & deploy process](https://github.com/mozilla/testpilot/blob/master/docs/development/hosting.md). Need to file a Bugzilla bug to request & track the work.

### PR DESCRIPTION
**Pull request by lmorchard**
* Add data/** to .eslintignore
* Add web-ext-artifacts/ to .gitignore
* Bump addon to v1.0.9 (it took 9 tries to get deployment to work)
* Ensure the correct signed .xpi gets renamed to signed-addon.xpi

Fixes #102.

### CODE REVIEW

**Comment by lmorchard** [comment]
Assuming this passes tests, I'm probably going to approve & merge myself just to get it into master and then force-push to production to test a "release"

**Comment by lmorchard** [comment]
Oh, actually I *can't* approve & merge my own PR. So... :man_shrugging:

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "snoozetabs",
   "description": "An add-on to let you snooze your tabs for a while.",
   "id": "snoozetabs@mozilla",
-  "version": "1.0.0",
+  "version": "1.0.9",
```
Comment (lmorchard) [inline package.json]: Pending a fix for #110, we need to update this version in both `package.json` and `manifest.json` to ensure the deploy process works. AMO will throw an error on signing if we attempt a push to `production` branch with a previously-signed version number - which is why we had to increment through v1.0.0 to v1.0.9 while troubleshooting 😞

**Comment by bwinton** [comment]
(Go for it, Les! 😃)

**Pull request merged**